packages/wallet/src/services/TransactionsTracker.ts

```diff
@@ -160,6 +160,7 @@ export const createAddressTransactionsProvider = ({
                 );
                 const duplicateTransactions =
                   lastStoredTransaction && intersectionBy(localTransactions, newTransactions, (tx) => tx.id);
```

Comment:
intersectionBy will be O(n*m)

Reply:
`newTransactions` are transactions starting from the `lastStoredTransaction` blockNo, up to the tip.
So to detect a rollback it is sufficient to check if `lastStoredTransaction` is the first transaction in `newTransactions`. There is no need for intersectionBy between `localTransactions` and `newTransactions`.
```diff
-                const duplicateTransactions =
-                  lastStoredTransaction && intersectionBy(localTransactions, newTransactions, (tx) => tx.id);
+                const lastTxRolledBack =
+                  lastStoredTransaction && newTransactions[0].id !== lastStoredTransaction.id;
```